runtime/lua/vim/_defaults.lua

```diff
@@ -215,173 +215,126 @@ do
 
   --- vim-unimpaired style mappings. See: https://github.com/tpope/vim-unimpaired
   do
+    ---@param lhs string
+    ---@param rhs function
+    ---@param desc string
+    local function create_unimpaired_mapping(lhs, rhs, desc)
+      vim.keymap.set('n', lhs, function()
+        local _, err = pcall(rhs) ---@type any, string
+        if err then
+          vim.api.nvim_err_writeln(err)
+        end
+      end, { desc = desc })
+    end
+
     -- Quickfix mappings
-    vim.keymap.set('n', '[q', function()
+    create_unimpaired_mapping('[q', function()
       vim.cmd.cprevious({ count = vim.v.count1 })
-    end, {
-      desc = ':cprevious',
-    })
-
-    vim.keymap.set('n', ']q', function()
+    end, ':cprevious')
```

Comment:
I think instead of creating multiple function to implement this we can should do:

```lua
vim.keymap.set('n', '[q', '<cmd>cprevious<cr>')
```

I don't think we even need desc since the output of `map [q` with this is:

```
n  [q          * <Cmd>cprevious<CR>
```

Reply:
This won’t work with counts. And will get ugly for some of the other mappings.

We already discussed this so not sure why it’s being brought up again. There’s nothing wrong with using functions here. It’s more explicit, better typed, and doesn’t suffer the obscure workarounds and edge cases of the Vimscript commands.

Look at vim-unimpaired’s source code if you want to see what the Vimscript implementations look like.

Reply:
> This won’t work with counts. And will get ugly for some of the other mappings.

Ok fair enough.

> We already discussed this so not sure why it’s being brought up again.

It was talked about here https://github.com/neovim/neovim/pull/28525#discussion_r1781850866 but I don't think that really covers this fully.

> There’s nothing wrong with using functions here. It’s more explicit, better typed, and doesn’t suffer the obscure workarounds and edge cases of the Vimscript commands.

I was only suggesting that using `rhs` as a single command name (no expression) would be cleaner and remove several layers of code (`create_unimpaired_mapping -> pcall -> vim.cmd -> nvim_cmd`). Obviously, I forgot about counts, so this doesn't work.

> Look at vim-unimpaired’s source code if you want to see what the Vimscript implementations look like.

Done. And you are right.